core/src/main/java/org/apache/iceberg/TableScanContext.java

```diff
@@ -60,6 +61,7 @@ final class TableScanContext {
     this.planExecutor = null;
     this.fromSnapshotInclusive = false;
     this.metricsReporter = new LoggingMetricsReporter();
+    this.branch = null;
```

Comment:
I think this should rather be `ref` or `referenceName`. Also I think it would be great to get #5982 / #5985 in before this PR